tensorflow_addons/image/filters.py

```diff
@@ -204,3 +204,57 @@ def median_filter2d(
         output = tf.cast(median, image.dtype)
         output = img_utils.from_4D_image(output, original_ndims)
         return output
+
+
+def gaussian_filter2d(
+    image: FloatTensorLike,
+    sigma: FloatTensorLike,
+    kernel_size: int,
+    padding: str = "REFLECT",
+    constant_values: TensorLike = 0,
+) -> FloatTensorLike:
+    """
+    This function is responsible for having Gaussian Blur. It takes the image as input, computes a gaussian-kernel
+    which follows normal distribution then convolves the image with the kernel.Presently it works only on
+    grayscale images.
+    Args:
+    image: A tensor of shape
+        (batch_size, height, width, channels)
+        (NHWC), (batch_size, channels, height, width)(NCHW).
+    sigma:A constant of type float64. It is the standard deviation of the normal distribution.
+          The more the sigma, the more the blurring effect.
+          G(x,y)=1/(2*3.14*sigma**2)e^((x**2+y**2)/2sigma**2)
+    kSize:It is the kernel-size for the Gaussian Kernel. kSize should be odd.
+          A kernel of size [kSize*kSize] is generated.
+    """
+
+    if sigma == 0:
+        raise ValueError("Sigma should not be zero")
+    if padding not in ["REFLECT", "CONSTANT", "SYMMETRIC"]:
+        raise ValueError("Padding should be REFLECT, CONSTANT, OR SYMMETRIC")
+
```

Comment:
Can you implement the `img_utils.to_4D_image` function that is used in the other filters. Unfortunately we can't merge a single channel only function because it's so likely that it would be used for more than just that.

Reply:
Done